server/market/bookrouter.go

```diff
@@ -57,25 +59,52 @@ func (bua bookUpdateAction) String() string {
 		return "newEpoch"
 	case matchProofAction:
 		return "matchProof"
+	case suspendAction:
+		return "suspend"
 	default:
 		return ""
 	}
 }
 
-// bookUpdateSignal combines a bookUpdateAction with data for which the action
+// updateSignal combines an updateAction with data for which the action
 // applies.
-type bookUpdateSignal struct {
-	action     bookUpdateAction
-	order      order.Order
+type updateSignal struct {
+	action updateAction
+	data   interface{} // sigData* type
+}
+
+func (us updateSignal) String() string {
+	return us.action.String()
+}
+
+type sigDataOrder struct {
+	order    order.Order
+	epochIdx int64
+}
+
+type sigDataBookedOrder sigDataOrder
+type sigDataUnbookedOrder sigDataOrder
+type sigDataEpochOrder sigDataOrder
```

Comment:
This seemed fine to me, but the linter says the order and `epochIdx` fields of `sigDataOrder` are unused.  That's clearly a false positive, but it might be an indication that this should be done differently. Embedding perhaps.